bin/format.js

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+
+const articles = findArticles(path.join(__dirname, '..'));
+const changes = {};
+const rules = [
+    null,
+    rule_1_boundaryWhitesapce,
+    rule_2_atxHeaders,
+    rule_3_noTabs,
+    rule_4_smartCharacters
+];
+
+const ruleArgIndex = process.argv.findIndex(a => a === '-r' || a === '--rule') + 1;
+const rulesToCheck = ruleArgIndex === 0 ? null : process.argv[ruleArgIndex].split(',').map(r => parseInt(r));
+
+articles.forEach(article => format(article));
+
+let changeCountTotal = 0;
+let changedFilesTotal = [];
+
+for (let i = 0; i < rules.length; i++) {
+    if (changes[i] === undefined)
+        continue;
+
+    const changeCount = Object.values(changes[i]).reduce((sum, c) => sum + c, 0);
+    const changedFiles = Object.entries(changes[i]).filter(c => c[1] > 0).map(c => c[0]);
+    const indent = ' '.repeat(4 - i.toString().length);
```

Comment:
This variable is only declared once, just put it at the declaration it represents.